include/boost/numeric/ublas/tensor/type_traits.hpp

```diff
@@ -0,0 +1,28 @@
+//
+//  Copyright (c) 2018-2020, Cem Bassoy, cem.bassoy@gmail.com
+//  Copyright (c) 2019-2020, Amit Singh, amitsingh19975@gmail.com
+//
+//  Distributed under the Boost Software License, Version 1.0. (See
+//  accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+//  The authors gratefully acknowledge the support of
+//  Google
+//
+
+#ifndef BOOST_UBLAS_TENSOR_TYPE_TRAITS_HPP
+#define BOOST_UBLAS_TENSOR_TYPE_TRAITS_HPP
+
+#include <boost/numeric/ublas/tensor/detail/type_traits_static_extents.hpp>
+#include <boost/numeric/ublas/tensor/detail/type_traits_static_strides.hpp>
+
+#include <boost/numeric/ublas/tensor/detail/type_traits_dynamic_extents.hpp>
+#include <boost/numeric/ublas/tensor/detail/type_traits_dynamic_strides.hpp>
+
+#include <boost/numeric/ublas/tensor/detail/type_traits_fixed_rank_extents.hpp>
+#include <boost/numeric/ublas/tensor/detail/type_traits_fixed_rank_strides.hpp>
+
+#include <boost/numeric/ublas/tensor/detail/type_traits_tensor.hpp>
+#include <boost/numeric/ublas/tensor/detail/storage_traits.hpp>
```

Comment:
Please have the following headers due to separation of concerns:
`type_traits_extents.hpp` that includes `type_traits` for `static_extents`, `dynamic_extents`, `fixed_rank_extents`
`type_traits_strides.hpp` that includes `type_traits` for `static_strides`, `dynamic_strides`, `fixed_rank_strides`
`type_traits_tensor.hpp` 
`type_traits_storage.hpp`

Reply:
I am not sure if we should put 
`type_traits_extents.hpp` 
`type_traits_strides.hpp`
` type_traits_tensor.hpp`
and 
` type_traits_storage.hpp` 
into one `type_traits` header.

Reply:
I wanted it like std `type_traits.hpp` which contains every type of trait. If you want separate traits I will decouple them.

Reply:
should I put into separate folder `type_traits`? This will make it easier to maintain and make better folder structure which we are missing for clean and maintainable code.

Reply:
Okay, what about this:
Let's have:
`details/type_traits_extents.hpp`
`details/type_traits_strides.hpp`
`details/type_traits_tensor.hpp`
`details/type_traits_storage.hpp`
and include them in
`type_traits.hpp`
Inside our code, we should only use `details` to have lesser inclusions.
Users of our code can access `type_traits`.

Reply:
This is much better.

Reply:
Then let's go with this version.